### PR TITLE
21977 Implemented Pending list

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#21977

**This is another incremental code change for the subject ticket.**

*Description of changes:*
- app version = 7.3.14
- now store pending bootstrap item to Pendings list
- added "pendings" store state, action, getter
- added pending conditionals to dashboard v-cards
- refactored PendingList.vue
- implemented PendingTemplate.vue
- implemented ContinuationIn.vue
- misc cleanup

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-683-s9ucczd2.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-683-s9ucczd2.web.app

File: src/App.vue [hunk 1]
```diff
@@ -687,7 +689,25 @@ export default class App extends Mixins(
...
+    const filing = response.filing as TaskTodoIF
+    // NB: these were already validated in storeBootstrapItem()
+    const header = filing.header
+    const data = filing[header.name]
+
+    // set addresses
+    this.storeAddresses({ data: data.offices || [] })
+
+    // set parties
+    this.storeParties({ data: { parties: data.parties || [] } })
```
Comment (severinbeauvais) [inline src/App.vue:701]: Although this bootstrap filing isn't complete, we  want to show the addresses and parties (both disabled) on the page, ie:

image

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-683-s9ucczd2.web.app

File: src/components/Dashboard/PendingList.vue
Comment (severinbeauvais) [inline src/components/Dashboard/PendingList.vue:1]: Left side is code taken from TodoList.vue. Right side is modeled after FilingHistoryList.vue. Way cleaner!

**Review by ArwenQin — APPROVED** [review]
LGTM!

File: src/components/Dashboard/PendingList.vue
Comment (severinbeauvais) [inline src/components/Dashboard/PendingList.vue:1]: Left side is code taken from TodoList.vue. Right side is modeled after FilingHistoryList.vue. Way cleaner!

File: src/components/Dashboard/PendingList/PendingTemplate.vue
Comment (severinbeauvais) [inline src/components/Dashboard/PendingList/PendingTemplate.vue:1]: This is the template for pending items.

Actual items (eg, ContinuationIn.vue) use this template and can override any of the slots.

File: src/components/Dashboard/PendingList/filings/ContinuationIn.vue [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    :index="index"
+  >
+    <template #body>
+      <div class="body-2">
+        <p class="mb-0">
+          This {{ entityType }} Continuation Application is paid and awaiting staff review. We will
+          notify you of the results by email. You can also come back here to check on the progress.
+        </p>
+      </div>
+    </template>
```
Comment (severinbeauvais) [inline src/components/Dashboard/PendingList/filings/ContinuationIn.vue:14]: This filing only needs to override the default (empty) body.

File: src/components/Dashboard/PendingList/filings/DefaultFiling.vue
Comment (severinbeauvais) [inline src/components/Dashboard/PendingList/filings/DefaultFiling.vue:1]: Here's a default (fallback) filing that uses the template and doesn't override any slots.

File: src/components/Dashboard/TodoList.vue [hunk 3]
```diff
@@ -69,7 +69,7 @@
           :class="{'invalid-section ml-n1 pl-1 rounded-0': showInvalidSection(item)}"
         >
           <div class="list-item">
-            <div class="todo-label">
+            <div class="todo-label pt-1">
```
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList.vue:72]: This (and line 2307) is a very subtle change but it better aligns the View Details button and the Continue In button.

Before:

image

After:

image

File: src/components/EntityInfo/EntityHeader.vue [hunk 4]
```diff
@@ -129,22 +131,14 @@ export default class EntityHeader extends Vue {
...
-    if (this.isContinuationInTodo) {
-      return this.getTasks[0]?.task.filing.displayName
-    }
-    // *** TODO: what needs to be done here for pending continuation bootstrap filing?
-    if (this.isContinuationInFiling) {
-      return this.getFilings[0]?.displayName
-    }
+    if (this.isContinuationInTodo) return this.getTasks[0]?.task.filing.displayName
+    if (this.isContinuationInPending) return this.getPendingsList[0].displayName
+    if (this.isContinuationInFiling) return this.getFilings[0]?.displayName
```
Comment (severinbeauvais) [inline src/components/EntityInfo/EntityHeader.vue:141]: ^^ get Display Name from correct list

Example when filing is in Pending list:

image

File: src/interfaces/root-state-interface.ts [hunk 5]
```diff
@@ -49,6 +49,8 @@ export interface RootStateIF {
   parties: Array<PartyIF>
   recordsAddress: OfficeAddressIF
   registeredAddress: OfficeAddressIF
-  tasks: Array<ApiTaskIF> // "tasks" data from API
+  // *** TODO: declare a type for pendings
+  pendings: Array<any> // pending list from bootstrap filing
```
Comment (severinbeauvais) [inline src/interfaces/root-state-interface.ts:53]: I'll do this in my next PR as I keep working on the ticket.

File: src/interfaces/root-state-interface.ts [hunk 6]
```diff
@@ -49,6 +49,8 @@ export interface RootStateIF {
   parties: Array<PartyIF>
   recordsAddress: OfficeAddressIF
   registeredAddress: OfficeAddressIF
-  tasks: Array<ApiTaskIF> // "tasks" data from API
+  // *** TODO: declare a type for pendingsList
+  pendingsList: Array<any> // pendings list from bootstrap filing
```
Comment (severinbeauvais) [inline src/interfaces/root-state-interface.ts:53]: I'll declare a type for this in my next PR as I keep working on the ticket.

File: tests/unit/PendingList.spec.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import Vue from 'vue'
+import Vuetify from 'vuetify'
+import { mount } from '@vue/test-utils'
+import PendingList from '@/components/Dashboard/PendingList.vue'
+
+Vue.use(Vuetify)
+const vuetify = new Vuetify({})
+
+// *** TODO: implement this
```
Comment (severinbeauvais) [inline tests/unit/PendingList.spec.ts:9]: I'll do this in my next PR as I keep working on the ticket.

File: src/components/Dashboard/PendingList/PendingTemplate.vue
Comment (severinbeauvais) [inline src/components/Dashboard/PendingList/PendingTemplate.vue:1]: This is the template for pending items.

Actual items (eg, ContinuationIn.vue, below) use this template and can override any of the slots.

File: src/components/Dashboard/PendingList/filings/DefaultFiling.vue
Comment (severinbeauvais) [inline src/components/Dashboard/PendingList/filings/DefaultFiling.vue:1]: Here's a default (fallback) filing that uses the template and doesn't override any slots.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-683-s9ucczd2.web.app

File: src/interfaces/root-state-interface.ts (hunk 6 above)
Comment (ketaki-deodhar) [reply src/interfaces/root-state-interface.ts:53]: Sounds good

**Review by ketaki-deodhar — APPROVED** [review]
Look good Sev!

**Pull request merged**